include/rdma/fi_eq.h

```diff
@@ -74,8 +72,8 @@ struct fid_wait {
 };
 
 struct fi_mutex_cond {
-	pthread_mutex_t		*mutex;
-	pthread_cond_t		*cond;
+	struct pthread_mutex_t	*mutex;
+	struct pthread_cond_t	*cond;
```

Comment:
i applied another type of forward declaration:
```
struct pthread_mutex_t;
struct pthread_cond_t;

struct fi_mutex_cond {
       pthread_mutex_t>*mutex;
       pthread_cond_t> *cond;
};
```
which better fit to C standard. patch to sock_wait.c is reverted